### ISSUE TITLE
Feedback - how would a screen reader verbalize a code block

### ISSUE THREAD

**Issue by rjekic** [WordPress/Learn#3283]
## Type of feedback
//dev

## Description

> General question for the WP Community: how would a screen reader verbalize a code block like this (as an example):
> 
> <?php
> $args = array(
>     'posts_per_page' => 5,
>     'post_type'      => 'post',
> );
> 
> $query = new WP_Query($args);
> 
> if ($query->have_posts()) :
>     while ($query->have_posts()) : $query->the_post();
> ?>
>         <h2><?php the_title(); ?></h2>
>         <p><?php the_excerpt(); ?></p>
> <?php
>     endwhile;
> else :
>     echo 'No posts found.';
> endif;
> 
> // Reset post data
> wp_reset_postdata();
> ?>
> 
> Is that even a concern? Would the WCAG flag [learn.wordpress.org](http://learn.wordpress.org/) if this kind of coding sample couldn't be articulated in a way that makes the context clear, ie. "this is a sample of PHP code?" 

 _Originally posted by @frereenfant-cmd in [#3273](https://github.com/WordPress/Learn/issues/3273#issuecomment-3225862567)_

**Comment by rjekic** [comment]
@frereenfant-cmd

Unfortunately, the situation regarding accessibility on our Learn content is not so great, and we will work on it in the future.

Here is the quick video I made that shows how a screen reader (NVDA on Windows) reads the code snippet created using **code** and **pre** tags. 

https://drive.google.com/file/d/1YmaVNRNHnK4WXIOG-DabF1wX5p8W8Zdq/view?usp=sharing 
(sorry for the bad video quality, I am not an expert in screen casting/recording with OBS on Windows :) )

As you can see and hear, the HTML code is read correctly - all parentheses, signs, and symbols are read correctly. On the other hand, PH has just partial support - parentheses are ignored, as far as I can tell, the exclamation is not read also...

I can just try to imagine how hard it is for people with visual impairment to read and learn from content that a screen reader reads like this. I am not sure how to help here, who to contact, or what should be changed, but I am here to help. We must do our best to create an environment that can be used equally by anyone.

cc @jonathanbossenger @devmuhib009

**Comment by jonathanbossenger** [comment]
A good place to start is to find out how code should be formatted correctly to work with screen readers. Once we know that, we can implement whatever we need in the theme. What are other learn platforms that offer code snippets doing to remain A11y compliant?

**Comment by rjekic** [comment]
@jonathanbossenger @frereenfant-cmd

During the WordCamp Vrsac that was held last weekend, I had the pleasure of meeting one of the accessibility experts and WordCamp speakers, Lazar Bulatovic. We tested a couple of Learning lessons and other learning platforms in and out of the WordPress ecosystem, and this is what we found:
- The code should be created using **pre** and **code** tags (we already use that)
- The code snippets reading depends on the screen reader settings. The NVDA reader that I used in the example above must be set to read all symbols (the shortcut for that is **Insert + P**, which must be pressed until the reader announces "use all symbols")
- The overall Learn WordPress page accessibility is good. We have to work on some small changes and check the general semantic structure of the page. But we are not violating WCAG guidelines.

We are open to suggestions from all people who have some sort of impairment and have trouble accessing pages on the learn.wordpress.org, so we can leave this issue open.

cc @devmuhib009

**Comment by rjekic** [comment]
I think it is time to close this issue. We got the answer regarding this issue. Since this issue is related to screen reader settings and not the content or code, and we could not initiate a discussion, we can close the issue and reopen it later if needed.
